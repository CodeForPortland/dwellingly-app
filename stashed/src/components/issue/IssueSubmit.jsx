import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../icon/Icon';

const IssueSubmit = ({ issueSent, issueSentError, onAcceptingSentSummary }) => (
  <div className={!issueSent ? "newIssueSent error" : "newIssueSent success"}>
    {!issueSentError ? <Icon icon="checkbox" /> : <Icon icon="close" />}
    <p className="issueStatusText">{issueSent ? "Sent!" : "Error!"}</p>
    <p className="issueStatusMessage">Message status goes here. Lorem ipsum dolar set</p>
    {!issueSentError ?
    <button
      type="button"
      className="btn btn--strong summaryButton"
      onClick={onAcceptingSentSummary}>
      Ok
    </button> :
    <button
      type="button"
      className="btn btn--strong btn--urgent summaryButton"
      onClick={onAcceptingSentSummary}>
      Retry
    </button>}
  </div>
)

IssueSubmit.propTypes = {
  issueSent: PropTypes.bool.isRequired,
  issueSentError: PropTypes.bool.isRequired,
  onAcceptingSentSummary: PropTypes.func.isRequired
}

export default IssueSubmit;
