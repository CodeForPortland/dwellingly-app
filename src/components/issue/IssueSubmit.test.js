import React from 'react';
import renderer from 'react-test-renderer';
import IssueSubmit from './IssueSubmit';

describe('IssueSubmit Component', () => {
  it('renders checkbox icon if issueSent=true', () => {
    const props = {
      issueSent: true,
    }
    const IssueSubmitComponent = renderer.create(<IssueSubmit {...props}/>).toJSON();
    expect(IssueSubmitComponent).toMatchSnapshot();
  });
  it('renders "Sent" message if issueSent=true', () => {
    const props = {
      issueSent: true,
    }
    const IssueSubmitComponent = renderer.create(<IssueSubmit {...props}/>).toJSON();
    expect(IssueSubmitComponent).toMatchSnapshot();
  });
  it('renders "Ok" button if issueSent=true and issueSentError=false', () => {
    const props = {
      issueSent: true,
      issueSentError: false,
    }
    const IssueSubmitComponent = renderer.create(<IssueSubmit {...props}/>).toJSON();
    expect(IssueSubmitComponent).toMatchSnapshot();
  });
  it('renders close icon if issueSent=false', () => {
    const props = {
      issueSent: false,
    }
    const IssueSubmitComponent = renderer.create(<IssueSubmit {...props}/>).toJSON();
    expect(IssueSubmitComponent).toMatchSnapshot();
  });
  it('renders "Error" message if issueSent=false', () => {
    const props = {
      issueSent: false,
    }
    const IssueSubmitComponent = renderer.create(<IssueSubmit {...props}/>).toJSON();
    expect(IssueSubmitComponent).toMatchSnapshot();
  });
  it('renders "Retry" button if issueSent=false & issueSentError=true', () => {
    const props = {
      issueSent: false,
      issueSentError: true,
    }
    const IssueSubmitComponent = renderer.create(<IssueSubmit {...props}/>).toJSON();
    expect(IssueSubmitComponent).toMatchSnapshot();
  });
});
