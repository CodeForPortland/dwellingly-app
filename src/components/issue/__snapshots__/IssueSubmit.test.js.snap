// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSubmit Component renders "Error" message if issueSent=false 1`] = `
<div
  className="newIssueSent error"
>
  <span
    className="icon icon--checkbox"
  >
    <svg
      viewBox="0 0 50 50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M25 50a25 25 0 1 1 0-50 25 25 0 0 1 0 50zm0-48a23 23 0 1 0 0 46 23 23 0 0 0 0-46z"
      />
      <path
        d="M22.1 35.2l-8.1-8 1.4-1.4 6.7 6.6L38 16.5l1.4 1.4z"
      />
    </svg>
  </span>
  <p
    className="issueStatusText"
  >
    Error!
  </p>
  <p
    className="issueStatusMessage"
  >
    Message status goes here. Lorem ipsum dolar set
  </p>
  <button
    className="btn btn--strong summaryButton"
    type="button"
  >
    Ok
  </button>
</div>
`;

exports[`IssueSubmit Component renders "Ok" button if issueSent=true and issueSentError=false 1`] = `
<div
  className="newIssueSent success"
>
  <span
    className="icon icon--checkbox"
  >
    <svg
      viewBox="0 0 50 50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M25 50a25 25 0 1 1 0-50 25 25 0 0 1 0 50zm0-48a23 23 0 1 0 0 46 23 23 0 0 0 0-46z"
      />
      <path
        d="M22.1 35.2l-8.1-8 1.4-1.4 6.7 6.6L38 16.5l1.4 1.4z"
      />
    </svg>
  </span>
  <p
    className="issueStatusText"
  >
    Sent!
  </p>
  <p
    className="issueStatusMessage"
  >
    Message status goes here. Lorem ipsum dolar set
  </p>
  <button
    className="btn btn--strong summaryButton"
    type="button"
  >
    Ok
  </button>
</div>
`;

exports[`IssueSubmit Component renders "Retry" button if issueSent=false & issueSentError=true 1`] = `
<div
  className="newIssueSent error"
>
  <span
    className="icon icon--close"
  >
    <svg
      viewBox="0 0 50 50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M27.8 25L48.4 4.4c.8-.8.8-2 0-2.8a2 2 0 0 0-2.8 0L25 22.2 4.4 1.6a2 2 0 0 0-2.8 0 2 2 0 0 0 0 2.8L22.2 25 1.6 45.5a2 2 0 0 0 0 2.8c.4.4.9.6 1.4.6.5 0 1-.2 1.4-.6L25 27.8l20.6 20.6c.4.4.9.6 1.4.6.5 0 1-.2 1.4-.6.8-.8.8-2 0-2.8L27.8 25z"
      />
    </svg>
  </span>
  <p
    className="issueStatusText"
  >
    Error!
  </p>
  <p
    className="issueStatusMessage"
  >
    Message status goes here. Lorem ipsum dolar set
  </p>
  <button
    className="btn btn--strong btn--urgent summaryButton"
    type="button"
  >
    Retry
  </button>
</div>
`;

exports[`IssueSubmit Component renders "Sent" message if issueSent=true 1`] = `
<div
  className="newIssueSent success"
>
  <span
    className="icon icon--checkbox"
  >
    <svg
      viewBox="0 0 50 50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M25 50a25 25 0 1 1 0-50 25 25 0 0 1 0 50zm0-48a23 23 0 1 0 0 46 23 23 0 0 0 0-46z"
      />
      <path
        d="M22.1 35.2l-8.1-8 1.4-1.4 6.7 6.6L38 16.5l1.4 1.4z"
      />
    </svg>
  </span>
  <p
    className="issueStatusText"
  >
    Sent!
  </p>
  <p
    className="issueStatusMessage"
  >
    Message status goes here. Lorem ipsum dolar set
  </p>
  <button
    className="btn btn--strong summaryButton"
    type="button"
  >
    Ok
  </button>
</div>
`;

exports[`IssueSubmit Component renders checkbox icon if issueSent=true 1`] = `
<div
  className="newIssueSent success"
>
  <span
    className="icon icon--checkbox"
  >
    <svg
      viewBox="0 0 50 50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M25 50a25 25 0 1 1 0-50 25 25 0 0 1 0 50zm0-48a23 23 0 1 0 0 46 23 23 0 0 0 0-46z"
      />
      <path
        d="M22.1 35.2l-8.1-8 1.4-1.4 6.7 6.6L38 16.5l1.4 1.4z"
      />
    </svg>
  </span>
  <p
    className="issueStatusText"
  >
    Sent!
  </p>
  <p
    className="issueStatusMessage"
  >
    Message status goes here. Lorem ipsum dolar set
  </p>
  <button
    className="btn btn--strong summaryButton"
    type="button"
  >
    Ok
  </button>
</div>
`;

exports[`IssueSubmit Component renders close icon if issueSent=false 1`] = `
<div
  className="newIssueSent error"
>
  <span
    className="icon icon--checkbox"
  >
    <svg
      viewBox="0 0 50 50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M25 50a25 25 0 1 1 0-50 25 25 0 0 1 0 50zm0-48a23 23 0 1 0 0 46 23 23 0 0 0 0-46z"
      />
      <path
        d="M22.1 35.2l-8.1-8 1.4-1.4 6.7 6.6L38 16.5l1.4 1.4z"
      />
    </svg>
  </span>
  <p
    className="issueStatusText"
  >
    Error!
  </p>
  <p
    className="issueStatusMessage"
  >
    Message status goes here. Lorem ipsum dolar set
  </p>
  <button
    className="btn btn--strong summaryButton"
    type="button"
  >
    Ok
  </button>
</div>
`;
